kraken/litmus/common_litmus.py

```diff
@@ -0,0 +1,92 @@
+    yaml_pod = yaml.full_load(pods_status)
+    logging.info('len of pod items' + str(len(yaml_pod['items'])))
+    for pod in yaml_pod['items']:
+        return pod['status']['phase']
+
+
+# Install litmus and wait until pod is running
+def install_litmus():
+    runcommand.invoke("kubectl apply -f "
+                      "https://litmuschaos.github.io/litmus/litmus-operator-v1.8.2.yaml")
+
+    # wait until pods are running
+
+    ready_status = get_pod_ready_status()
+    logging.info("first ready status for pod" + str(ready_status))
+    while ready_status != "Running":
+        time.sleep(10)
+        logging.info("ready status for pod" + str(ready_status))
+        ready_status = get_pod_ready_status()
+        if "Error" in ready_status:
+            logging.error("Litmus pod had error creating")
+            sys.exit(1)
+
+
+# Check status of experiment
+def check_experiment(engine_name, experiment_name, namespace):
+    chaos_engine = runcommand.invoke("kubectl describe chaosengines/%s -n %s | grep "
+                                     "Status" % (engine_name, namespace))
+    engine_status = chaos_engine.split(": ")[-1].strip()
+    max_tries = 2
+    engine_counter = 0
+    while engine_status != "Running" and engine_status != "Completed":
+        time.sleep(10)
+        logging.info("Sleeping waiting for engine to start running")
+        chaos_engine = runcommand.invoke("kubectl describe chaosengines/%s -n %s | grep "
+                                         "Status" % (engine_name, namespace))
+        engine_status = chaos_engine.split(": ")[-1].strip()
+        if engine_counter >= max_tries:
+            logging.error("Chaos engine took longer than 5 minutes to be running or complete")
+            return False
+        engine_counter += 1
+        # need to see if error in run
+        if "not found" in engine_status:
+            return False
+
+    if not chaos_engine:
+        return False
+    chaos_result = runcommand.invoke("kubectl describe chaosresult %s"
+                                     "-%s -n %s | grep Verdict" %
+                                     (engine_name, experiment_name, namespace))
+    result_counter = 0
+    status = chaos_result.split(": ")[1].strip()
+    while status == "Awaited":
+        logging.info("Waiting for chaos result to finish waiting, sleeping 10 seconds")
+        time.sleep(10)
+        chaos_result = runcommand.invoke("kubectl describe chaosresult %s"
+                                         "-%s -n %s | grep Verdict" %
+                                         (engine_name, experiment_name, namespace))
+        status = chaos_result.split(": ")[1].strip()
+        if result_counter >= max_tries:
+            logging.error("Chaos results took longer than 5 minutes to get a final result")
+            return False
+        result_counter += 1
+        if "not found" in status:
+            return False
+
+    if status == "Pass":
```

Comment:
It would be nice to take into consideration/results also litmus probes state, in future.